File: ts/lsp/servers/diagnosticls.ts
import { ServerConfigType } from '../constants';
import { npm } from '../installers/npm';
import { getServerDir } from '../utils';

const serverName = "diagnosticls"
export const config: ServerConfigType = {
  name: serverName,
  installer: () => npm("diagnostic-languageserver", getServerDir(serverName)),
  defaultOptions: {
    cmd: [getServerDir(serverName) + '/node_modules/.bin/diagnostic-languageserver', '--stdio']
  }
};

File: lua/lspinstall/servers/diagnosticls.lua
local config = require"lspinstall/util".extract_config("diagnosticls")
config.default_config.cmd[1] = "./node_modules/.bin/diagnostic-languageserver"

return vim.tbl_extend('error', config, {
  install_script = [[
  ! test -f package.json && npm init -y --scope=lspinstall || true
  npm install diagnostic-languageserver@latest
  ]]
})
